### ISSUE TITLE
ForeignKeyAttribute on skip navigations fails to remove by-convention join table from model

### ISSUE THREAD

**Issue by janschreier** [dotnet/efcore#27990]
With code created by dbScaffold (using global tool 6.0.4) in EF Core 6.0.4 I get the following error:
> Exception thrown: 'Microsoft.Data.Sqlite.SqliteException' in Microsoft.EntityFrameworkCore.Relational.dll
An unhandled exception of type 'Microsoft.Data.Sqlite.SqliteException' occurred in Microsoft.EntityFrameworkCore.Relational.dll
SQLite Error 1: 'table "MotorartMotorbauart" already exists'.

I think the error must exist for at least some time, but I can't tell exactly which version introduced it.

Here's a sample program creating that error: https://github.com/janschreier/EfCoreInMemoryBug
the dbContext code-is from a larger project and i tried to strip it down as much as possible while still receiving the same error. So it looks a bit akward.

`OnModelCreating()` is also called twice which is unexpected to me but then I never had a breakpoint on it so I'm not sure if that's okay or not

EF Core version: 6.0.4
Database provider:  Microsoft.EntityFrameworkCore.Sqlite
Target framework:  net6.0.4
Operating system: win 10x64
IDE:  Visual Studio 2022  17.1.6

**Comment by AndriySvyryd** [comment]
Probably a duplicate of https://github.com/dotnet/efcore/issues/25797

**Comment by bricelam** [comment]
@ajcvickers Do you think this was fixed by PR #29794?

**Comment by ajcvickers** [comment]
@bricelam Still throws with 8.0 preview 2

**Comment by christianlevesque** [comment]
We were having a similar problem at work when setting up a unit test database, and we thought maybe it was related to this issue. However, in the process of diagnosing our error, we also figured out what's happening with the reproduction code. I know it's old, but it's still open, so I figured I would drop a quick comment.

The reproduction code is configuring the relationships twice - once in `DbContext.OnModelCreating()` and again in the models using attributes. I submitted a PR back into the reproduction project showing that by removing the attribute configuration, the code runs as expected.
